tycho-apitools-plugin/src/main/java/org/eclipse/tycho/apitools/ApiAnalysisMojo.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.equinox.p2.metadata.IInstallableUnit;
+import org.eclipse.pde.api.tools.internal.provisional.ApiPlugin;
+import org.eclipse.pde.api.tools.internal.provisional.problems.IApiProblem;
+import org.eclipse.pde.core.target.ITargetPlatformService;
+import org.eclipse.sisu.equinox.EquinoxServiceFactory;
+import org.eclipse.tycho.IDependencyMetadata.DependencyMetadataType;
+import org.eclipse.tycho.ReactorProject;
+import org.eclipse.tycho.TargetEnvironment;
+import org.eclipse.tycho.TargetPlatform;
+import org.eclipse.tycho.core.TychoProject;
+import org.eclipse.tycho.core.osgitools.DefaultReactorProject;
+import org.eclipse.tycho.core.utils.TychoProjectUtils;
+import org.eclipse.tycho.extras.pde.ListDependenciesMojo;
+import org.eclipse.tycho.osgi.TychoServiceFactory;
+import org.eclipse.tycho.osgi.adapters.MavenLoggerAdapter;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult;
+import org.eclipse.tycho.p2.resolver.facade.P2Resolver;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolverFactory;
+import org.eclipse.tycho.p2.target.facade.TargetPlatformConfigurationStub;
+
+/**
+ * Performs a PDE-API Tools analysis of this project.
+ */
+@Mojo(name = "analyse", defaultPhase = LifecyclePhase.VERIFY, threadSafe = true, requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME, requiresOnline = true)
+// TODO: or later?
+public class ApiAnalysisMojo extends AbstractMojo {
+
+	@Component(hint = TychoServiceFactory.HINT)
+	private EquinoxServiceFactory equinox;
+
+	@Requirement(hint = "connect")
+	private EquinoxServiceFactory serviceFactory;
+
+	@Component
+	private Logger plexusLogger;
+
+	@Parameter(property = "project", readonly = true)
+	private MavenProject project;
+
+	@Component(role = TychoProject.class)
+	private Map<String, TychoProject> projectTypes;
+
+	@Parameter(property = "tycho.apitools.analyse.baseline")
+	private List<URL> baselines; // FIXME: make required? or use String? Just like CompareWithBaselineMojo?
+
+	@Parameter(defaultValue = "false", property = "tycho.apitools.analyse.skip")
+	private boolean skip;
+
+	@Component(role = APIAnalyser.class)
+	private APIAnalyser apiAnalyser;
+
+	@Override
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			getLog().info("Skipped");
+			return;
+		}
+		if (new File(project.getBasedir(), JarFile.MANIFEST_NAME).isFile()) { // TODO: consider non PDE bundles
```

Comment:
I think a more general approach would be to take a look at the project artifact and check if:

1. it is a jar
2. it is an OSGi Bundle